src/hip/hip_build_utils.cpp

```diff
@@ -39,6 +39,71 @@
 MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_HIP_VERBOSE)
 MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_HIP_DUMP)
 
+#if MIOPEN_OFFLINE_COMPILER_PATHS_V2
+
+// Flags to hold Relative Directory Path
+// for each Compiler Flags from ROCM Install Path
+// This flag Paths are expected to be deprecated/modified
+// in upcoming MAJOR Releases.
+#define MIOPEN_CLANG_REL_PATH "llvm/bin/clang"
+#define MIOPEN_OCL_REL_PATH "bin/clang-ocl"
+#define MIOPEN_CPPCLANG_REL_PATH "llvm/bin/clang++"
+#define MIOPEN_OFFLOADBUNDLER_REL_PATH "llvm/bin/clang-offload-bundler"
+
+// Function to generate the MIOPEN Compiler Path Value using
+// ROCm Base Install Path fetched using getROCmInstallPath()
+// This approach depends on the getROCmInstallPath() provided by rocm-core
+// This flag Paths are expected to be deprecated/modified in upcoming MAJOR Releases.
+static std::string generateCompilerPathValue(const char* relativePath)
+{
+    char* rocmPath   = nullptr;
+    unsigned int len = 0;
+    std::string compilerPathValue;
+    if(nullptr != relativePath)
+    {
+        PathErrors_t ret = getROCmInstallPath(&rocmPath, &len);
+        if(PathSuccess == ret)
+        {
+            if(PATH_MAX_LEN > (len + strlen(relativePath)))
+            {
+                compilerPathValue = std::string(rocmPath) + std::string(relativePath);
+            }
+            free(rocmPath);
```

Comment:
[Recommendation]
I do not believe that getROCmInstallPath() uses `malloc()` to allocate memory for rocmPath. If it does, then it must be reworked:
```C++
// If ptr == nullptr, writes the size of memory (required to hold ROCm
// installation path) into *len.
//
// If ptr != nullptr, writes ROCm installation path (ASCIZ string) into memory pointed
// by ptr, assuming that the size of memory is not smaller than *len.
// If *len is smaller than size of the string being written, returns error.
// The behavior is undefined if *len is larger than the actual size of
// the available memory.
PathErrors_t
    getROCmInstallPath(
        char* const ptr,
        size_t* const len);
```
The caller owns the memory.

Alternatively, if the rocm-core library instance holds a C string internally AND it is guaranteed that that C string lifetime is not less than the lifetime of the caller, then the caller can avoid allocating/deallocating of memory:
```C++
// Writes a pointer to the ROCm installation path (ASCIZ string) into *ptr.
PathErrors_t
    getROCmInstallPath(
        char** ptr);
```

Reply:
Usually, such an API is called twice if a buffer of unknown size is required. First, the function returns how much memory needs to be allocated through parameters. The second time is called to store bytes in the buffer. For example, see the behaviors of the `std::getenv_s()` function. https://en.cppreference.com/w/c/program/getenv.
The call to `std::getenv_s` with a null pointer for value and zero for values determines the buffer size required to hold the entire result.

Reply:
In general, the function should return, e.g., ERROR_BUFFER_TOO_SMALL, in case the provided memory block is too small to hold the value, and at the same time, return the required size through parameter. Then, the caller needs to reallocate and call the function again.

Reply:
See here https://github.com/ROCm/MIOpen/blob/b44f182b731532b5c32eb37a2df16b18efe4d94f/src/expanduser.cpp#L204